File: frontend/src/types/news.ts
import { Category } from '@/store/newsStore';

export interface Sentiment {
    polarity: number;
    label: 'positive' | 'negative' | 'neutral';
}

export interface Article {
    title: string;
    description?: string;
    url: string;
    urlToImage?: string;
    publishedAt: string;
    source?: string;
    author?: string;
    sentiment?: Sentiment;
    category?: Category;
}

export interface NewsResponse {
    articles: Article[];
    total: number;
} 

File: backend/src/models/news.py
from pydantic import BaseModel, Field, HttpUrl
from typing import List, Optional

class Source(BaseModel):
    id: Optional[str] = None
    name: str

class Article(BaseModel):
    source: Source
    author: Optional[str] = None
    title: str
    description: Optional[str] = None
    url: HttpUrl
    urlToImage: Optional[HttpUrl] = None
    publishedAt: str
    content: Optional[str] = None

class ArticleResponse(BaseModel):
    status: str
    totalResults: int
    articles: List[Article] 